src/decaylanguage/dec/dec.py

```diff
@@ -422,15 +422,15 @@ def _add_decays_to_be_copied(self):
         # match name -> position in list self._parsed_decays
         name2treepos = {
             t.children[0].children[0].value: i
-            for i, t in enumerate(self._parsed_decays)
+            for i, t in enumerate(self._parsed_decays)  # type: ignore
```

Comment:
This (and _most_ of the ignores below) are due to the fact that a multi-state class is a bad design (I've written many of them). Basically, this class is only valid if you call `.parse`; so every method should have a `assert self._parsed_decays is not None` at the front, or something similar. It's better if there were two classes (parsed/unparsed) or parse was part of the constructor, etc.

Reply:
I did a check on the parsing in several places but for sure not everywhere. This is something we can revisit. Thanks.